packages/core-data/src/actions.js

```diff
@@ -696,15 +696,17 @@ export const __experimentalSaveSpecifiedEntityEdits = (
 /**
  * Returns an action object used in signalling that Upload permissions have been received.
  *
- * @param {boolean} hasUploadPermissions Does the user have permission to upload files?
+ * @deprecated since WP 5.9, use receiveUserPermission instead.
  *
  * @return {Object} Action object.
  */
-export function receiveUploadPermissions( hasUploadPermissions ) {
+export function receiveUploadPermissions() {
+	deprecated( "wp.data.dispatch( 'core' ).receiveUploadPermissions", {
+		since: '5.9',
+	} );
+
 	return {
-		type: 'RECEIVE_USER_PERMISSION',
-		key: 'create/media',
-		isAllowed: hasUploadPermissions,
+		type: 'DO_NOTHING',
```

Comment:
We should still internally point to `receiveUserPermission` to keep backwards compatibility for now, no?

Reply:
I agree. I think we just need a deprecation message here.

Reply:
Something like this ? 

<img width="848" alt="Screenshot 2021-12-21 at 9 23 32 PM" src="https://user-images.githubusercontent.com/46275049/146959902-6a0c4426-d1ab-4db5-b8aa-f26d66e52018.png">

Reply:
Yes, but you should bring back the `hasUploadPermissions` param in the function's signature.

Reply:
oh yes.